doc/specs/#4999 - Improved keyboard handling in Conpty.md

```diff
@@ -0,0 +1,452 @@
+    - For ex, <kbd>Ctrl+space</kbd> - should we send `NUL` or
+      \<chosen_encoding's version of ctrl+space>
+* If there's a scenario where Windows Terminal might _not_ be connected to a
+  conpty, then how does conpty enable \<chosen_encoding>?
+* Is the goal "Full `INPUT_RECORD` fidelity" or "Make the above scenarios work"?
+  - One could imagine having the Terminal special-case the above keys, and send
+    the xterm modifyOtherKeys sequences just for those scenarios.
+    - This would _not_ work for <kbd>shift</kbd> all by itself.
+  - In my _opinion_, "just making the above work" is a subset of "full
+    INPUT_RECORD", and inevitably we're going to want "full INPUT_RECORD"
+
+The goal we're trying to achieve is communicating `INPUT_RECORD`s from the
+terminal to the client app via conpty. This isn't supposed to be a \*nix
+terminal compatible communication, it's supposed to be fundamentally Win32-like.
+
+
+Keys that we definitely need to support, that don't have unique VT sequences:
+* <kbd>Ctrl+Space</kbd> ([#879], [#2865])
+* <kbd>Shift+Enter</kbd> ([#530])
+* <kbd>Ctrl+Break</kbd> ([#1119])
+* <kbd>Ctrl+Alt+?</kbd> ([#3079])
+* <kbd>Ctrl</kbd>, <kbd>Alt</kbd>, <kbd>Shift</kbd>,  (without another keydown/up) ([#3608], [#4334], [#4446])
+
+## Solution Design
+
+### Inspiration
+
+The design we've settled upon is one that's highly inspired by a few precedents:
+* `Application Cursor Keys (DECCKM)` is a long-supported VT sequences which a
+  client application can use to request a different input format from the
+  Terminal. This is the DECSET sequence `^[[?1h`/`^[[?1l` (for enable/disable,
+  respectively). This changes the sequences sent by keys like the Arrow keys
+  from a sequence like `^[[A` to `^[OA` instead.
+* The `kitty` terminal emulator uses a similar DECSET sequence for enabling
+  their own input format, which they call ["full mode"]. Similar to DECCKM, this
+  changes the format of the sequences that the terminal should send for keyboard
+  input. Their "full mode" contains much more information when keys are pressed
+  or released (though, less than a full `INPUT_RECORD` worth of data). Instead
+  of input being sent to the client as a CSI or SS3 sequence, this `kitty` mode
+  uses "Application Program-Command" (or "APC) sequences , prefixed with `^[_`.
+* [iTerm2](https://www.iterm2.com/documentation-escape-codes.html) has a region
+  of OSC's that they've carved for themselves all starting with the same initial
+  parameter, `1337`. They then have a number of commands that all use the second
+  parameter to indicate what command specific to iTerm2 they're actually
+  implementing.
+
+### Requesting `win32-input-mode`
+
+An application can request `win32-input-mode` with the following OSC sequence
+
+```
+  ^[ ] 1000 ; 1 ; Ps ST
+
+         Ps: Whether to enable or disable win32-input-mode. If omitted, defaults to '0'.
+             0: Disable win32-input-mode
+             1: Enable win32-input-mode
+```

```

Comment:
Is there any reason why this isn't just a private `SM`/`RM` mode, the same as `DECKPM`, and Kitty's keymode mode? One advantage is that it can then be queried with `DECRQM` if you need to know whether the mode had actually been applied or not.

Reply:
I briefly considered it, but I guess I decided against it because I thought it would be easier to not conflict with an existing sequence by using an OSC. I definitely agree thoguh, a private mode does seem more _stylistically_ correct, plus I forgot about the ability to query those modes. 

I'll move this to `CSI ? 9001 h/l`, that doesn't look taken 😄